batch/batch/driver/instance_collection/pool.py

```diff
@@ -323,28 +323,36 @@ async def regions_to_ready_cores_mcpu_from_estimated_job_queue(self) -> List[Tup
   SELECT scheduling_iteration, user_idx, n_regions, regions_bits_rep, CAST(COALESCE(SUM(cores_mcpu), 0) AS SIGNED) AS ready_cores_mcpu
   FROM (
     SELECT {user_idx} AS user_idx, batch_id, job_id, cores_mcpu, always_run, n_regions, regions_bits_rep,
-      ROW_NUMBER() OVER (ORDER BY batch_id, always_run DESC, -n_regions DESC, regions_bits_rep, job_id ASC) DIV {share} AS scheduling_iteration
+      ROW_NUMBER() OVER (ORDER BY batch_id, job_group_id, always_run DESC, -n_regions DESC, regions_bits_rep, job_id ASC) DIV {share} AS scheduling_iteration
```

Comment:
job_group_id is not part of the unique identifier of a job. Why should it be included in this ordering and numbering? AFAIK, this ordering and numbering is meant to convert the jobs table into kind of queue where we repeatedly look at the first `share`'s worth of jobs.

Reply:
Because the scheduler first finds all running job groups in order and then selects jobs based on the job group. Therefore, we need to make sure the autoscheduler is looking at jobs in order of job group as well.